listenbrainz/db/stats.py

```diff
@@ -52,11 +53,10 @@ def insert_user_stats(user_id, artists, recordings, releases, artist_count):
              artist_count (int): the total number of artists listened to by the user
     """
 
+    artist_range = artists['range']
+    del artists['range']
```

Comment:
modifying the dict in place can lead to problems.

```
# suppose we have a variable called artist_stats
db.insert_user_stats(artist_stats)

# get range for some other use
range = artists['range']
```
This will error out, but it's not obvious to the user of the insert function.

It is a good idea to keep functions pure as much as possible, that is take input -> give output, without modifying the input. We should probably make a copy here.